AIMSICD/src/main/java/com/SecUpwN/AIMSICD/utils/Device.java

```diff
@@ -159,29 +147,24 @@ public int getPhoneID() {
         return mPhoneID;
     }
 
+    private String getSimInformation(String simInfo, String errorMsg) {
+        try {
+            if(simInfo != null && !simInfo.isEmpty())
```

Comment:
i have no idea about the catch clause. i didn't put it there see: https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/753#issuecomment-172175981. i think it should be removed. but then something will presumably crash on some devices if the comment is correct.
and no, neither =! nor return will ever throw anything. isEmpty could of cause throw an npe if =! null isn't there but that's not really the point.